src/simulation/dynamics/facetSRPDynamicEffector/facetSRPDynamicEffector.h

```diff
@@ -0,0 +1,68 @@
+#include "architecture/_GeneralModuleFiles/sys_model.h"
+#include "architecture/utilities/bskLogging.h"
+#include "architecture/messaging/messaging.h"
+#include "architecture/msgPayloadDefC/SpicePlanetStateMsgPayload.h"
+#include "simulation/dynamics/_GeneralModuleFiles/stateData.h"
+#include "architecture/utilities/avsEigenSupport.h"
+#include "architecture/utilities/avsEigenMRP.h"
+
+/*! @brief Spacecraft Geometry Data */
+typedef struct {
+  std::vector<double> facetAreas;                                   //!< [m^2] Vector of facet areas
+  std::vector<double> facetSpecCoeffs;                              //!< Vector of facet spectral reflection optical coefficients
+  std::vector<double> facetDiffCoeffs;                              //!< Vector of facet diffuse reflection optical coefficients
+  std::vector<Eigen::Vector3d> facetNormals_B;                      //!< Vector of facet normals expressed in B frame components
+  std::vector<Eigen::Vector3d> facetLocationsPntB_B;                //!< [m] Vector of facet locations wrt point B in B frame components
+}FacetedSRPSpacecraftGeometryData;
+
+/*! @brief Faceted Solar Radiation Pressure Dynamic Effector */
+class FacetSRPDynamicEffector: public SysModel, public DynamicEffector
+{
+public:                                                             
+    FacetSRPDynamicEffector();                                      //!< The module constructor
+    ~FacetSRPDynamicEffector();                                     //!< The module destructor
+    void linkInStates(DynParamManager& states);                     //!< Method for giving the effector access to the hub states
+    void computeForceTorque(double integTime, double timeStep);     //!< Method for computing the SRP force and torque about point B
+    void Reset(uint64_t CurrentSimNanos);                           //!< Reset method
+    void UpdateState(uint64_t CurrentSimNanos);                     //!< Method for updating the effector states
+    void WriteOutputMessages(uint64_t CurrentClock);                //!< Method for writing the output messages
+    void addFacet(double area, double specCoeff, double diffCoeff, Eigen::Vector3d normal_B, Eigen::Vector3d locationPntB_B);  //!< Method for adding facets to the spacecraft geometry structure
+
+    ReadFunctor<SpicePlanetStateMsgPayload> sunInMsg;               //!< Sun spice ephemeris input message
+    uint64_t numFacets;                                             //!< Total number of spacecraft facets
+    Eigen::Vector3d r_SN_N;                                         //!< [m] Position vector of the Sun with respect to the inertial frame
+    StateData *hubPosition;                                         //!< [m] Hub inertial position vector
+    StateData *hubSigma;                                            //!< MRP attitude of the hub B frame with respect to the inertial frame
+    BSKLogger bskLogger;                                            //!< BSK Logging
```

Comment:
:small_red_triangle_down: There's already a BSKLogger in the class from which this class is inheriting. You don't need this member variable. Delete.

Reply:
Done.